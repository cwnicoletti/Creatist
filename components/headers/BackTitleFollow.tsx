import React from "react";
import { Image, Platform, Text, View, ActivityIndicator } from "react-native";
import { HeaderButtons, Item } from "react-navigation-header-buttons";
import { useAppSelector } from "../../hooks";
import Back from "../headers_components/Back";
import Follow from "../headers_components/Follow";
import Title from "../headers_components/Title";
import IoniconsHeaderButton from "../UI/header_buttons/IoniconsHeaderButton";
import SimpleLineIconsHeaderButton from "../../components/UI/header_buttons/SimpleLineIconsHeaderButton";

const BackTitleFollow = ({ navigation }) => {
  const darkModeValue = useAppSelector((state) => state.user.darkMode);
  const isfollowing = navigation.getParam("isfollowing");
  const isLoading = navigation.getParam("isLoading");
  const ExhibitUId = navigation.getParam("ExhibitUId");
  const exploredExhibitUId = navigation.getParam("exploredExhibitUId");
  const followFn = navigation.getParam("followFn");
  const unfollowFn = navigation.getParam("unfollowFn");

  return (
    <View>
      <View
        style={{
          padding: 23,
          backgroundColor: darkModeValue ? "black" : "white",
        }}
      />
      <View style={{ flexDirection: "row" }}>
        <Back
          darkModeValue={darkModeValue}
          navigation={navigation}
          HeaderButton={IoniconsHeaderButton}
          HeaderButtons={HeaderButtons}
          Item={Item}
          View={View}
          Platform={Platform}
        />
        <Title
          View={View}
          Image={Image}
          Text={Text}
          darkModeValue={darkModeValue}
        />
        <Follow
          darkModeValue={darkModeValue}
          navigation={navigation}
          HeaderButton={SimpleLineIconsHeaderButton}
          HeaderButtons={HeaderButtons}
          Item={Item}
          View={View}
          ActivityIndicator={ActivityIndicator}
          Platform={Platform}
          isfollowing={isfollowing}
          isLoading={isLoading}
          ExhibitUId={ExhibitUId}
          exploredExhibitUId={exploredExhibitUId}
          followFn={followFn}
          unfollowFn={unfollowFn}
        />
      </View>
    </View>
  );
};

export default BackTitleFollow;
