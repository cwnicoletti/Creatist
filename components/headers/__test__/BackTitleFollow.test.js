import React from "react";
import renderer from "react-test-renderer";
import BackTitleFollow from "../BackTitleFollow";
import authReducer from "../../../store/reducers/auth";
import signupReducer from "../../../store/reducers/signup";
import userReducer from "../../../store/reducers/user";
import { Provider } from "react-redux";
import { applyMiddleware, combineReducers, createStore } from "redux";
import ReduxThunk from "redux-thunk";

const rootReducer = combineReducers({
  signup: signupReducer,
  auth: authReducer,
  user: userReducer,
});

const store = createStore(rootReducer, applyMiddleware(ReduxThunk));

test("matches previous snapshot and renders correctly", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BackTitleFollow navigation={{ getParam: jest.fn() }} />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
