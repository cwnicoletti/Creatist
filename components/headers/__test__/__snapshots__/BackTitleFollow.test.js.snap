// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches previous snapshot and renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "padding": 23,
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderBottomWidth": 1,
          "borderColor": "gray",
          "flex": 1,
          "padding": 10,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "justifyContent": "center",
            },
            Object {
              "marginRight": 5,
            },
          ]
        }
      >
        <RNGestureHandlerButton
          activeOpacity={0.3}
          borderless={true}
          collapsable={false}
          disabled={false}
          enabled={true}
          hitSlop={
            Object {
              "bottom": 5,
              "left": 5,
              "right": 5,
              "top": 5,
            }
          }
          onActiveStateChange={[Function]}
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          onPress={[Function]}
          rippleRadius={20}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <View>
            <Text />
          </View>
        </RNGestureHandlerButton>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderBottomWidth": 1,
          "borderColor": "gray",
          "flex": 6,
          "paddingBottom": 7,
          "paddingTop": 9,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontFamily": "CormorantUpright",
              "fontSize": 26,
            }
          }
        >
          ExhibitU
        </Text>
      </View>
    </View>
    <View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderBottomWidth": 1,
            "borderColor": "gray",
            "flex": 1,
            "marginLeft": 23,
            "marginRight": 23,
            "padding": 10,
          }
        }
      />
    </View>
  </View>
</View>
`;
